Add copy constructor to `RequestConfiguration{Descriptor}`

flobernd opened the pull request:
As titled.

What is the deal with `DisablePing`, `EnableHttpPipelining` in `RequestConfiguration` (both properties are not part of the interface) and `_enableHttpCompression` and `_maxRetryTimeout` in `RequestConfigurationDescriptor` (both fields don't have a corresponding setter-method)?

Is that by design or a leftover of the refactoring?

stevejgordon commented:
> As titled.
> 
> What is the deal with `DisablePing`, `EnableHttpPipelining` in `RequestConfiguration` (both properties are not part of the interface) and `_enableHttpCompression` and `_maxRetryTimeout` in `RequestConfigurationDescriptor` (both fields don't have a corresponding setter-method)?
> 
> Is that by design or a leftover of the refactoring?

I think some of those need cleaning/fixing up. `DisablePing`, I think, was renamed to `DisablePings` and `EnabledHttpPipelining` to ` `HttpPipeliningEnabled`. My guess is `_enableHttpCompression` and `_maxRetryTimeout` should have setter methods.

Mpdreamz reviewed (changes requested):
Echo'ing what Steve said, the RequestCongfiguration and RequestData need another pass to ensure conformity. 

Happy to take that on in another PR :)

For this PR can we add another test to ensure all properties get passed through similar to `ITransportConfiguration` ?

https://github.com/elastic/elastic-transport-net/blob/95ea6c6f4d6da04641bed5029cdcc8ae66e0558c/tests/Elastic.Transport.Tests/Configuration/TransportConfigurationTests.cs#L42

flobernd commented on src/Elastic.Transport/Components/Pipeline/RequestPipeline.cs:
```diff
@@ -58,7 +58,7 @@ private RequestConfiguration PingAndSniffRequestConfiguration
 				PingTimeout = PingTimeout,
 				RequestTimeout = PingTimeout,
 				Authentication = _requestData.AuthenticationHeader,
-				EnableHttpPipelining = _requestData.HttpPipeliningEnabled,
```
That property was not wired at all 😅

flobernd commented:
I removed the properties which are not present in the interface, added methods for the ones that ARE present in the interface but were not settable in the descriptor, and added some tests for the `RequestConfiguration{Descriptor}`.

stevejgordon reviewed (approved):
LGTM - thanks
